Shiny Data Ingest App (Work in progress)

LiamBurke24 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Bringing online most of the UI elements needed to register new data with BETY. To this end, I realized that the entire workflow had to be modularized and I took some steps to this end. This is not a final version of the app as I need to make some formal "Shiny Modules". But this is my best guess at how the final workflow will look. These changes are not final. They are not breaking, so this PR can be merged, or we can keep it open until I finish the formal modularization early this week. 

I have NOT updated CHANGELOG.md because these changes are not final. 

<!--- Please select appropriate Priority, Status,and Type labels-->
<!--- If you do not have permission to select labels please state which labels you would like -->

## Description
<!--- Describe your changes in detail -->
I broke up the input record workflow (which consisted of  input record, dbfiles record, and format record steps). Doing so allows me to include these steps in the "local_upload" and d1 Download workflows. In this way, users can follow the workflow without changing tabs -- greatly increasing efficiency and making implementing a cohesive/ reactive workflow more possible. This workflow is only included in the local_upload tab since I have yet to formally modularize the UI and server functions (I discovered the need to do this too late for this PR). Nonetheless, this PR will serve as an example of what the workflow should end up looking like. and should demonstrate the functionality. 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Without modularizing the app, the workflow is rather user hostile. By putting all the steps to ingesting data on one scree, this allows the user to cross reference steps and to see the process through much more easily. From an implementation standpoint, modularizing the input workflow allows me to let inputs from various steps cascade into later steps (i.e. filenames, filepaths). A final version of this workflow may even allow for the creation of multiple records for multiple files using shiny's `renderUI` functionality. 

<!--- If it fixes an open issue, please link to the issue here. -->

## Review Time Estimate
<!---When do you want your code reviewed by?-->
- [ ] Immediately
- [x] Within one week
- [ ] When possible
## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--this template is from https://www.talater.com/open-source-templates/#/page/99-->

mdietze commented on shiny/Data-Ingest/ui_files/input_record_ui.R:
```diff
@@ -118,6 +120,7 @@ fluidRow(title = "New Input",
          height = '150px'
        ),
        actionButton("createFormatRecord", label = "Create Format Record"),
+       p("* Denotes a Required Field"),
```
Not completely following the logic of which fields are listed as required, as many of the starred fields seem optional unless other choices are made in specific combinations. More generally, some screenshots would be nice to better see what this actually looks like

LiamBurke24 replied:
Ok! Noted. It would be great to discuss what fields are required for each input. I put in the asterisks assuming that some of these fields would be required (based on my best judgement). Beyond that, I'm not sure I understand the logic either. And OF COURSE! I figured you would be able to view the changes once they were merged, but good point about the screen shots! I'll post those now.

LiamBurke24 replied:
Also just noticed that this is the two thousandth pull request YAY!

LiamBurke24 commented:
Homepage:
![screenshot-2018-7-9 pecan data ingest](https://user-images.githubusercontent.com/29068693/42460958-e74c0ae0-8364-11e8-93b5-1f136f1da0b1.png)

DataONE upload:
![screenshot-2018-7-9 pecan data ingest 1](https://user-images.githubusercontent.com/29068693/42461203-8ecef1ce-8365-11e8-94aa-f9fcadf4aaa8.png)

Upload local files: 
![screenshot-2018-7-9 pecan data ingest 2](https://user-images.githubusercontent.com/29068693/42461253-a6deb57e-8365-11e8-9c01-5ecea269e3da.png)

LiamBurke24 commented:
Thank you for all of this wonderful feedback (I just saw this now for some reason). I don't have time this second to address it now, but I will either later tonight or in the morning! Thank you! In the meantime, I just pushed changes modularizing each of the steps in the ingest workflow. It is a REALLY cool feature that I think will help out many of the shiny apps in PEcAn. For instance, other apps could now easily implement those parts of the ingest progress natively if they so choose. Thanks again.

LiamBurke24 commented:
Thank you for all the wonderful feedback today. Ok, so I was able to add two of the points that we discussed today and will hopefully get the rest tomorrow. I added a `create New Format` button that renders the formats tab. I also added a link to the bety website instead of the `new Format` box.  

![screenshot_2018-07-10 pecan data ingest](https://user-images.githubusercontent.com/29068693/42550245-42a69568-8496-11e8-9fca-aac7b40cf58b.png)

![screenshot_2018-07-10 pecan data ingest 1](https://user-images.githubusercontent.com/29068693/42550249-4a069f9c-8496-11e8-9551-7c10561d2048.png)

LiamBurke24 commented:
Thank you! Doh, whoops. I thought you just wanted me to make the time stuff not required. (I figured start/ end dates were required) I'll change that really quick.
